blockchain/contracts/HelloWorld.sol

```diff
@@ -1,52 +1,124 @@
+    using AddrArrayLib for AddrArrayLib.Addresses;
     address payable nftOwner;
     address public buyer;
 
+    uint256[] public availableNFTs;
 
-    uint256[] public nfts;
+    uint256 public nftsLength;
 
     struct NFTOwnership {
         address owner;
         uint256 nftId;
-        uint droptime;
+        uint256 dropTime;
+        uint256 dropHash;
+        address reservedFor;
     }
 
     mapping(uint256 => NFTOwnership) public nftOwnerships;
 
+    // Used to track which addresses have joined the drop
+    // address[] private joinedUsers;
+    AddrArrayLib.Addresses joinedUsers;
 
     constructor() {
         buyer == msg.sender;
         nftOwner = payable(0xC77787e364E3420c0609a249F18De47430900f0C);
     }
 
-    // create a NFT
-    function mockNFT(uint _droptime) public returns (uint256) {
-        uint256 nftHash = getNFTidentifier();
+    function createDrop(uint256 _dropTime) public {
+        uint256 dropHash = generateRandomNumber(0);
+        uint256 i;
+        // I choose "3" as a value for testing reasons
+        for (i = 0; i < 3; i++) {
+            mockNFT(_dropTime, dropHash, i + 1);
+        }
+    }
+
+    // create NFTs for drop
+    function mockNFT(
+        uint256 _dropTime,
+        uint256 _dropHash,
+        uint256 _number
+    ) public {
+        uint256 nftHash = generateRandomNumber(_number);
         NFTOwnership memory nftOwnership;
         nftOwnership.nftId = nftHash;
         nftOwnership.owner = nftOwner;
-        nftOwnership.droptime = _droptime;
+        nftOwnership.reservedFor = nftOwner;
+        nftOwnership.dropTime = _dropTime;
+        nftOwnership.dropHash = _dropHash;
         nftOwnerships[nftHash] = nftOwnership;
-        nfts.push(nftHash);
-        return nftHash;
+        availableNFTs.push(nftHash);
+        nftsLength = availableNFTs.length;
+    }
+
+    function joinDrop() public {
+        // TODO: purchase limit of 5% of all available NFTs
+        require(
+            !joinedUsers.exists(msg.sender),
+            "The address you are calling from has already joined the drop."
+        );
+        // We have to make sure that not more users join the drop than we have NFTs available
+        require(joinedUsers.size() <= 10, "You cannot join the drop anymore.");
+        joinedUsers.pushAddress(msg.sender);
     }
 
-    function buyNFT(uint256 _price, uint256 _nftHash) public payable  {
-        require(nftOwnerships[_nftHash].droptime<=block.timestamp,"Droptime not yet reached!");
-        //string(abi.encodePacked("Drop has not started yet! ",((nftOwnerships[_nftHash].droptime-block.timestamp)/86400)," Days, ",((nftOwnerships[_nftHash].droptime-block.timestamp)/3600)," Hours,", ((nftOwnerships[_nftHash].droptime-block.timestamp)/60)," Minutes, and ", ((nftOwnerships[_nftHash].droptime-block.timestamp))," Seconds left.")));
+    function drop(uint256 _dropHash) public {
+        require(
+            nftOwnerships[_dropHash].dropTime <= block.timestamp,
```

Comment:
here dropHash is used as index to retriev Drop time. Results in Requirement not working, unless nftHash is used as Parameter. (Rest of function works as supposed, as anyways all available NFTs are to be dropped)

Reply:
nftHash is supposed to be used as a Parameter here to execute the correct drop

Reply:
What do you mean by "Results in Requirement not working, unless nftHash is used as parameter"?

Reply:
resolved with the newest commit :) I got rid of the dropHash altogether since we don't need it atm